sessions/recap-project-4/example-implementation/src/App.js

```diff
@@ -0,0 +1,81 @@
+import useLocalStorageState from "use-local-storage-state";
+import { useEffect } from "react";
+import "./App.css";
+import Form from "./components/Form/index.js";
+import List from "./components/List/index.js";
+import { nanoid } from "nanoid";
+
+function App() {
+  const [activities, setActivities] = useLocalStorageState("activities", {
+    defaultValue: [],
+  });
+  const [weather, setWeather] = useLocalStorageState("weather", {
+    defaultValue: null,
+  });
+  console.log(weather);
+
+  useEffect(() => {
+    async function fetchWeather() {
+      try {
+        const response = await fetch(
+          "https://example-apis.vercel.app/api/weather"
+        );
+        if (response.ok) {
+          const data = await response.json();
+          setWeather(data);
+        } else {
+          console.log("Something went wrong!");
+          setWeather({
+            temperature: "⁉️",
+            condition: "⁉️",
+            isGoodWeather: null,
+          });
+        }
+      } catch (error) {
+        console.log(error);
+      }
+    }
+    const timer = setInterval(() => {
+      fetchWeather();
+    }, 5000);
+
+    return () => {
+      clearInterval(timer);
+    };
+  }, [setWeather]);
+
+  const filteredActivities = activities.filter(
+    (activity) => activity.isGoodWeather === weather.isGoodWeather
+  );
+
+  function handleAddActivity(newActivity) {
+    setActivities([...activities, { id: nanoid(), ...newActivity }]);
+  }
+
+  function handleDeleteActivity(id) {
+    setActivities(activities.filter((activity) => activity.id !== id));
+  }
+
+  return (
+    <div className="App">
+      <h1 className="App__heading">
+        {weather.isGoodWeather === null ? (
+          "Ooops, looks like there is no weather at all!"
+        ) : (
+          <>
+            <span>{weather.condition}</span>
+            <span>{weather.temperature} °C</span>
+          </>
+        )}
+      </h1>
+      <List
```

Comment:
Do we not want to provide at least one default activity per weather condition?

Reply:
Well, I guess this is a good idea.